src/KafkaW/TopicSettings.h

```diff
@@ -1,17 +1,15 @@
 #pragma once
 
+#include "SettingsInterface.h"
 #include "logger.h"
 #include <librdkafka/rdkafka.h>
-#include <map>
-#include <string>
 
 namespace KafkaW {
 
-class TopicSettings {
-public:
+struct TopicSettings : public KafkaW::SettingsInterface {
   TopicSettings();
-  void applySettingsToRdKafkaConf(rd_kafka_topic_conf_t *conf);
+  virtual void apply() override{};
```

Comment:
This implementation doesn't do anything and isn't used. I suggest we get rid of `SettingsInterface` until we convert to the C++ API, as we can then use something like this for the interface: 

```
struct SettingsInterface {
  virtual void apply(RdKafka::Conf &KafkaConf) = 0;
  std::map<std::string, std::string> Configuration;
};

```